prereise/gather/winddata/hrrr/hrrr_api.py

```diff
@@ -0,0 +1,115 @@
+import requests
+from pandas import date_range
+
+from prereise.gather.winddata.hrrr.grib import GribRecordInfo
+from prereise.gather.winddata.hrrr.helpers import (
+    formatted_filename,
+    generate_index_url,
+    get_indices_that_contain_selector,
+)
+
+
+class HrrrApi:
+    """Class to interact with to get and download HRRR data. More information
+    about HRRR data can be found at
+    `this link <https://registry.opendata.aws/noaa-hrrr-pds/>`_
+    and
+    `this link <https://www.nco.ncep.noaa.gov/pmb/products/hrrr/>`_
+
+    :param prereise.gather.winddata.hrrr.downloader.Downloader downloader:
+    :param str base_url: url to download data from. Should take "dt", "product"
+        and "hours_forecasted" as format variables. See
+        prereise.gather.winddata.hrrr.constants for more detail
+    """
+
+    U_COMPONENT_FILTER = "UGRD:80 m above ground"
+    V_COMPONENT_FILTER = "VGRD:80 m above ground"
+
+    def __init__(self, downloader, base_url):
+        self.downloader = downloader
+        self.base_url = base_url
+
+    def _filename_url_iter(self, start_dt, end_dt, product):
+        """Iterates from a start datetime (inclusive) to a end datetime (exclusive)
+        at 1 hour steps, returning a filename as well as a url based
+        on each intermediary datetime
+
+        :param datetime.datetime start_dt: datetime to start at
+        :param datetime.datetime end_dt: datetime to end at
+        :param str product: info at `this link <https://www.nco.ncep.noaa.gov/pmb/products/hrrr/>`_
+
+        :return: (*Generator[tuple]*)
```

Comment:
Not sure this is the correct way of referencing a `iterator`? @rouille might have a better idea.

Reply:
Our convention is to only give the type of the returned variable (as type(variable) would return and when necessary be more specific in the description, I.e., elements of an iterable, indices and columns of a data frame, keys and values of a dictionary, etc.

Reply:
I think this would be the right way to specify the type in many cases, e.g. type hints or definitions could use the following
```
from typing import Generator

def foo(my_arg: Generator[int]):
    pass
```

Going with Ben's suggestion to use `type(x)` it would just be `generator`.